### DISCUSSION TITLE
Some questions about reflections.

### DISCUSSION THREAD

**Discussion by westtrade** [vlang/v#16836] [Questions and Answers] [answered]
Hi guys! I learned v reflections - and found next case which not covered by docs or examples:

We have two structures

```
struct Gun {
 name string
}

struct Character {
	name string
	gun Gun
}
```
I found that's we can walk around all fields in our types, but I don't found a way to walk recursive in nested structs. Does we have a language tools for make structures walks recursivelly?

**Comment by felipensp** [comment, marked as answer]
Hello. Take a look the sample at https://github.com/vlang/v/pull/16777

**Comment by enghitalo** [comment]
> Hi guys! I learned v reflections - and found next case which not covered by docs or examples:
> 
> We have two structures
> 
> ```
> struct Gun {
>  name string
> }
> 
> struct Character {
> 	name string
> 	gun Gun
> }
> ```
> I found that's we can walk around all fields in our types, but I don't found a way to walk recursive in nested structs. Does we have a language tools for make structures walks recursivelly?
> 
> 
> 
> 
> 
> 

Try use `$if field.typ is $Struct` or `$if field.typ is $Map`
